package/MDAnalysis/coordinates/GRO.py

```diff
@@ -311,15 +335,17 @@ def write(self, selection, frame=None):
                 "Alternatively these can be supplied as keyword arguments."
                 "".format(miss=', '.join(missing_topology)))
 
+        positions = selection.positions
+
         if self.convert_units:
             # Convert back to nm from Angstroms,
-            # inplace because coordinates is already a copy
-            self.convert_pos_to_native(coordinates)
+            # Not inplace because AtomGroup is not a copy
+            positions = self.convert_pos_to_native(positions, inplace=False)
             if has_velocities:
                self.convert_velocities_to_native(velocities) 
```

Comment:
if you don't want to modify the positions you shouldn't modify the velocities either.